### PR TITLE
Bulk Upload Accounts

### PR DESCRIPTION
**Pull request by calsupik**
Modal for bulk uploading password accounts

Utilize `PapaParse` for parsing CSV in browser: https://www.papaparse.com/

TODO:

- [x] Wire into contract to bulk add accounts to blockchain.

### CODE REVIEW

File: src/views/dashboards/PasswordManager/BulkPasswordAccountModal.vue [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                    | #[i.now-ui-icons.arrows-1_share-66] Upload CSV
+
+              div.mt-2.table-responsive
+                table.table.table-striped.table-bordered.m-0
+                  thead
+                    tr
+                      th Account Name
+                      th Username
+                      th Password
+                      th Additional Info
```
Comment (whatl3y) [inline src/views/dashboards/PasswordManager/BulkPasswordAccountModal.vue:37]: Can we add a "remove" column with a red X to remove the item if a user wants or something is not correct by mistake?

**Review by whatl3y — CHANGES_REQUESTED** [review]
This looks great!

In addition to the one comment about item removal, here are a few more things:

- Can we add a manual add form with 4 inputs to manually add an item to the list if a user wants?
- Can we add a checkbox that asks the user whether the CSV has headers or not so the first row isn't added if it does in fact have headers?
- Can we add a download template link? There are ways to create a file object and download it all on the browser without ever needing to interact with a server.
- I'd like to seamlessly support XLSX files if possible. I know users will want this and not have to convert for both bulk uploading accounts and bulk token sender.

File: src/views/dashboards/PasswordManager/BulkPasswordAccountModal.vue [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+  data() {
+    return {
+      needsToWriteDownPrivateKey: false,
+      uploadedAccounts: [],
+    };
+  },
+
+  computed: mapState({
+    cost: (state) => state.passwordManager.cost,
```
Comment (whatl3y) [inline src/views/dashboards/PasswordManager/BulkPasswordAccountModal.vue:95]: The cost is actually 50% cost * number of accounts.

**Comment by whatl3y** [comment]
One mroe thing

**Comment by whatl3y** [comment]
Also, there is a hard limit in the contract that requires a MINIMUM of 5 accounts to use the bulk upload service. Should ensure this limit is shown and the button is disabled until there are at least 5 accounts added.

File: src/views/dashboards/PasswordManager/BulkPasswordAccountModal.vue (hunk 2 above)
Comment (calsupik) [reply src/views/dashboards/PasswordManager/BulkPasswordAccountModal.vue]: 5790096

File: src/views/dashboards/PasswordManager/BulkPasswordAccountModal.vue (hunk 1 above)
Comment (calsupik) [reply src/views/dashboards/PasswordManager/BulkPasswordAccountModal.vue:37]: 5790096

**Comment by calsupik** [comment]
> This looks great!
> 
> In addition to the one comment about item removal, here are a few more things:
> 
> * Can we add a manual add form with 4 inputs to manually add an item to the list if a user wants? 
5790096

> * Can we add a checkbox that asks the user whether the CSV has headers or not so the first row isn't added if it does in fact have headers?
Right now I have added logic to check if the first rows first column contains "name" or "account name" and if it does, to pass the `headers` param when parsing file. This is not bulletproof, but thought it might be a smarted way instead of asking the end user to check a box if headers are present. Do we want to also add the checkbox?

> Also, there is a hard limit in the contract that requires a MINIMUM of 5 accounts to use the bulk upload service. Should ensure this limit is shown and the button is disabled until there are at least 5 accounts added.
5790096

**Pull request merged**